Tests/xSharePoint/xSharePoint.xSPSearchTopology.Tests.ps1

```diff
@@ -0,0 +1,307 @@
+                Add-Type -TypeDefinition "public class QueryProcessingComponent { public string ServerName { get; set; } public System.Guid ComponentId {get; set;}}"
+                Add-Type -TypeDefinition "public class IndexComponent { public string ServerName { get; set; } public System.Guid ComponentId {get; set;} public System.Int32 IndexPartitionOrdinal {get; set;}}"
+
+                $adminComponent = New-Object AdminComponent
+                $adminComponent.ServerName = $env:COMPUTERNAME
+                $adminComponent.ComponentId = [Guid]::NewGuid()
+
+                $crawlComponent = New-Object CrawlComponent
+                $crawlComponent.ServerName = $env:COMPUTERNAME
+                $crawlComponent.ComponentId = [Guid]::NewGuid()
+
+                $contentProcessingComponent = New-Object ContentProcessingComponent
+                $contentProcessingComponent.ServerName = $env:COMPUTERNAME
+                $contentProcessingComponent.ComponentId = [Guid]::NewGuid()
+
+                $analyticsProcessingComponent = New-Object AnalyticsProcessingComponent
+                $analyticsProcessingComponent.ServerName = $env:COMPUTERNAME
+                $analyticsProcessingComponent.ComponentId = [Guid]::NewGuid()
+
+                $queryProcessingComponent = New-Object QueryProcessingComponent
+                $queryProcessingComponent.ServerName = $env:COMPUTERNAME
+                $queryProcessingComponent.ComponentId = [Guid]::NewGuid()
+
+                $indexComponent = New-Object IndexComponent
+                $indexComponent.ServerName = $env:COMPUTERNAME
+                $indexComponent.IndexPartitionOrdinal = 0
+
+                return @($adminComponent, $crawlComponent, $contentProcessingComponent, $analyticsProcessingComponent, $queryProcessingComponent, $indexComponent)
+            }
+
+            Mock Get-SPEnterpriseSearchServiceInstance  {
+                return @{
+                    Server = @{
+                        Address = $env:COMPUTERNAME
+                    }
+                    Status = "Online"
+                }
+            }
+
+            $testParams = @{
+                ServiceAppName          = "Search Service Application"
+                Admin                   = @($env:COMPUTERNAME)
+                Crawler                 = @($env:COMPUTERNAME)
+                ContentProcessing       = @($env:COMPUTERNAME)
+                AnalyticsProcessing     = @($env:COMPUTERNAME)
+                QueryProcessing         = @($env:COMPUTERNAME)
+                IndexPartition          = @($env:COMPUTERNAME)
+                FirstPartitionDirectory = "I:\SearchIndexes\0"
+            }
+
+            It "returns true from the test method" {
+                Test-TargetResource @testParams | Should Be $true
+            }
+        }
+    }
+}
```

Comment:
Hey @BrianFarnhill 

Code seems to be good, rather long as expected, but seems ok. My only concern is, what if someone wanted to just 'ensure topology' from a supposedly existent Search Service App and that app doesn't exist, or if it exists and the topology is screwed to the point it  seems there is no topology.